source/deimos/openssl/opensslv.d

```diff
@@ -1,7 +1,99 @@
+/**
+ * Module to deal with the library version being used
+ *
+ * This library provide bindings for a wide range of OpenSSL versions,
+ * ranging from v0.9.x to v3.0.x. Some versions are not compatible with
+ * one another, either due to different ABI or different behavior,
+ * for example OpenSSL 1.0 requires initialization but later versions do not.
+ *
+ * While things tend to mostly work or error out while linking when the version
+ * the bindings assume and the actually C library version are too different,
+ * we prefer to try detecting the currently used version, and allow users
+ * to specify the version explicitly, before falling back to the latest bindings
+ */
 module deimos.openssl.opensslv;
 
 import deimos.openssl._d_util;
 
+version (DeimosOpenSSL_1_0_0)
+{
+    // https://www.openssl.org/news/changelog.html#openssl-100
+    // OpenSSL 1.0.0t was released 2015-12-03
+    public alias OpenSSLVersion = OpenSSLVersionTemplate!"1.0.0t";
+}
+else version (DeimosOpenSSL_1_0_1)
+{
+    // https://www.openssl.org/news/changelog.html#openssl-101
+    // OpenSSL 1.0.1u was released 2016-09-22
+    public alias OpenSSLVersion = OpenSSLVersionTemplate!"1.0.1u";
+}
+else version (DeimosOpenSSL_1_0_2)
+{
+    // https://www.openssl.org/news/changelog.html#openssl-102
+    // OpenSSL 1.0.2t was released 2019-09-10
+    public alias OpenSSLVersion = OpenSSLVersionTemplate!"1.0.2t";
+}
+else version (DeimosOpenSSL_1_1_0)
+{
+    // https://www.openssl.org/news/changelog.html#openssl-110
+    // OpenSSL 1.1.0l was released 2019-09-10
+    public alias OpenSSLVersion = OpenSSLVersionTemplate!"1.1.0l";
+}
+else version (DeimosOpenSSL_1_1_1)
+{
+    // https://www.openssl.org/news/changelog.html#openssl-111
+    // OpenSSL 1.1.1m was released 2021-12-14
+    public alias OpenSSLVersion = OpenSSLVersionTemplate!"1.1.1m";
+}
+else version (DeimosOpenSSL_3_0)
+{
+    // https://www.openssl.org/news/changelog.html#openssl-30
+    // OpenSSL 3.0.3 was released 2022-05-03
+    public alias OpenSSLVersion = OpenSSLVersionTemplate!"3.0.3";
+}
+else version (DeimosOpenSSL_3_1)
+{
+    // https://www.openssl.org/news/changelog.html#openssl-31
+    // OpenSSL 3.0.3 was released 2022-05-03
+    public alias OpenSSLVersion = OpenSSLVersionTemplate!"3.0.3";
+}
+else version (DeimosOpenSSLAutoDetect)
+{
+    import deimos.openssl.version_;
+
+    public alias OpenSSLVersion = OpenSSLVersionTemplate!OpenSSLTextVersion;
+}
+else
+{
+    // It was decided in https://github.com/D-Programming-Deimos/openssl/pull/66
+    // that we should fall back to the latest supported version of the bindings,
+    // should the user provide neither explicit version nor `DeimosOpenSSLAutoDetect`
+    public alias OpenSSLVersion = OpenSSLVersionTemplate!"1.1.0h";
```

Comment:
I think this should have the same version as the fallback one written by `generate_version`, so either this should be `1.1.0` or that one changed to `1.1.0h`.

Reply:
Actually according to the comment, shouldn't this be `3.0.3`?

I think we need to establish a policy of how to determine (and when to update) the default/fallback targeted OpenSSL version.

Maybe use the OpenSSL version of `/usr/lib/libssl.so` of the latest Ubuntu LTS?